LESSONS.md

```diff
@@ -0,0 +1,7 @@
+2020-08-18 08:49:56
+
+# Lessons Learned
+
+-   TypeSet vs TypeList
+-   Structure files vs single file per resource/data source
+-   Single API per resource/datasource
```

Comment:
I mean we kind of did. I think this can be read more like "Resources should have a 1:1 relationship with their associated part of the API". The example is any of the resources that use REST and/or NerdGraph are still a 1:1 relationship with their specific feature. The only resource that contradicts that is `alert_policy` I believe, which leveraged the policy REST endpoint and the policy_channel REST endpoint when it should only have been using the policy endpoint to adhere to the 1:1 relationship.